protocol/messenger_mailserver.go

```diff
@@ -1080,6 +1126,9 @@ func (m *Messenger) ConnectionChanged(state connection.State) {
 			m.logger.Error("failed to start datasync", zap.Error(err))
 		}
 	}
+	if !state.Offline {
+		m.asyncRequestAllHistoricMessages()
```

Comment:
I'm pretty sure this is already done somewhere 🤔

Reply:
here it is:
https://github.com/status-im/status-go/blob/9403475572978aa62da64c954d0d7defd7c58beb/protocol/messenger.go#L1006-L1009

Reply:
Also, looks like we need to reuse `m.config.codeControlFlags.AutoRequestHistoricMessages` in all of those places 🤔  This flag is set to false in some tests (e.g. to test manual requests to store nodes).

Don't know why I didn't put it directly into `asyncRequestAllHistoricMessages`, probably we can just move it inside this function.

Reply:
Took care of this here: 091855bba9d3f5ba0abd26be3e353af642432e6a